traffic_ops/traffic_ops_golang/deliveryservice/keys_test.go

```diff
@@ -20,214 +20,728 @@ package deliveryservice
+LbAOeqpvmHgLxzUNl2zWguv8HvcO7dPq0EPMAWYXygOrjQEt2CToISCH8X9Cmfpx
+eiWf8K0yDjOzCkidwnYhGA5g4MvvawRuhPvny9dCXZIz0esv7fZkaU01QzlzIhZ7
+iy8BD+DmvPKTeyT/APlYJDKpxHDio4DybmMCAwEAAaMjMCEwDwYDVR0TAQH/BAUw
+AwEB/zAOBgNVHQ8BAf8EBAMCAoQwDQYJKoZIhvcNAQELBQADggEBALBKBleeU18A
+nMREEVlK/IFwj79rJ+Mq89qonSixHgnocMYCbqthRZMY7SnQQo5p1ywKImZXxZ5z
+DiCnnTtal6I3N0g3XMgVdkViaRr3lHuDnYB8BdPnxgIhzQfEVWi/XT4CvAcrW43O
+WJ+SCf+UnGk5IAC2/4rHMEcCuzX54ftDC5lJR01lsHuTDdlk1LVqnX4ELlOeGaGP
+uiftf8XGEvkA/0aeYDEgAqe4JGZdRkh28NYf+jaPX13VgJ9rvrKejElp+0hrX3fp
+8GIGZLNzCEwusuoWnjQbaSp7vJGDZ5bNX8KnQ8MwBLmU+Z2E6RXm5JVsvGzkSNWL
++pkmRD85yuk=
 -----END CERTIFICATE-----
 `
 )
 
-func TestVerifyAndEncodeCertificate(t *testing.T) {
+func TestDecodePrivateKey_PKCS8_RSA_2048(t *testing.T) {
+
+	privateKey, cleanPemPrivateKey, err := decodePrivateKey(PrivateKey_PKCS8_RSA_2048)
+
+	if err != nil {
+		t.Errorf("Unexpected result: " + err.Error())
+	}
+
+	pBlock, remain := pem.Decode([]byte(cleanPemPrivateKey))
+
+	if pBlock == nil {
+		t.Error("can't decode cleaned private key pem block")
+		t.FailNow()
+	} else if len(remain) > 0 {
+		t.Error("remaining bytes after decode > 0. expected: 0")
+		t.FailNow()
+	}
+
+	if privateKey == nil {
+		t.Error("RSA private key is nil. expect: not nil")
+		t.FailNow()
+	}
+}
+
+func TestDecodePrivateKey_PKCS1_RSA_2048(t *testing.T) {
+
+	privateKey, cleanPemPrivateKey, err := decodePrivateKey(PrivateKey_PKCS1_RSA_2048)
 
+	if err != nil {
+		t.Errorf("Unexpected result: " + err.Error())
+	}
+
+	pBlock, remain := pem.Decode([]byte(cleanPemPrivateKey))
+
+	if pBlock == nil {
+		t.Error("can't decode cleaned private key pem block")
+		t.FailNow()
+	} else if len(remain) > 0 {
+		t.Error("remaining bytes after decode > 0. expected: 0")
+		t.FailNow()
+	}
+
+	if privateKey == nil {
+		t.Error("RSA private key is nil")
+		t.FailNow()
+	}
+}
+
+func TestDecodePrivateKey_RSA_Encrypted_And_BadData(t *testing.T) {
+
+	// Expected to fail on decode of encrypted pem private key
+	privateKey, _, err := decodePrivateKey(PrivateKey_ENCRYPTED_RSA_2048)
+	if err == nil && privateKey != nil {
+		t.Error("unexpected result: decoding of encrypted private key should have returned an error")
+		t.FailNow()
+	}
+
+	// Expected to fail.
+	privateKey, _, err = decodePrivateKey(BadKeyData)
+	if err == nil && privateKey != nil {
+		t.Error("unexpected result: decoding of bad private key data should have returned an error")
+		t.FailNow()
+	}
+}
+
+func TestVerifyAndEncodeCertificate_BadData(t *testing.T) {
 	// should fail bad base64 data
-	dat, _, err := verifyCertificate(BadData, "")
+	dat, pemEncodedKey, unknownAuth, err := verifyCertKeyPair(BadCertData, BadKeyData, "")
 	if err == nil {
 		t.Errorf("Unexpected result, there should have been a base64 decoding failure")
 	}
 
-	// should fail, can't verify self signed cert against this rootCA
-	dat, _, err = verifyCertificate(SelfSignedCertOnly, rootCA)
-	if err == nil {
+	_ = dat
+	_ = pemEncodedKey
+	_ = unknownAuth
```

Comment:
what's this doing?

Reply:
It's "using" the variables, because Go won't let you compile unused variables. I suggested removing them and making the function call itself `_, _, _, err := verifyCertKeyPair(` elsewhere.

Reply:
ohhh, okay. Makes sense.